build/templates/session.py.mako

```diff
@@ -87,6 +87,7 @@ class ${session_context_manager}(object):
 
 
 % endif
+% if config['use_locking']:
 # From https://stackoverflow.com/questions/5929107/decorators-with-parameters
 def ivi_synchronized(f):
     def aux(*xs, **kws):
```

Comment:
Unrelated to your change but I noticed you have a decorator not using [`wraps`](https://docs.python.org/3/library/functools.html#functools.wraps)

Reply:
You can open a separate issue for this.

Reply:
To a point.

I am not going to go create a separate PR to fix a typo or to fix an error report. At some point you add too much overhead and it becomes easier to just leave it be.

For these particular things, I knew what to do to fix the error report. For wraps I will need to read up on them and figure out how to use them, and then probably write a test. It is a bigger thing.